grails-app/services/org/broadinstitute/orsp/QueryService.groovy

```diff
@@ -786,22 +786,87 @@ class QueryService implements Status {
+                "i.summary summary,  " +
+                "i.reporter reporter,  " +
+                "ur.display_name reporter_name, " +
+                "i.approval_status,  " +
+                "i.update_date updated,  " +
+                "i.expiration_date expirationDate, " +
+                "iep.name, " +
+                "iep.value ," +
+                "u.display_name " +
+                "FROM issue i  " +
+                "LEFT JOIN issue_extra_property iep  " +
+                "ON (iep.project_key = i.project_key  " +
+                "AND iep.name IN ('pm','pi','collaborator')) " +
+                "LEFT JOIN user u ON (u.user_name = iep.value) " +
+                "LEFT JOIN user ur ON (ur.user_name = i.reporter) " +
+                "WHERE i.id IN (:issueIds) " +
+                "AND i.deleted != 1"
+
         SessionFactory sessionFactory = grailsApplication.getMainContext().getBean('sessionFactory')
         final session = sessionFactory.currentSession
         final SQLQuery sqlQuery = session.createSQLQuery(query)
-        final results = sqlQuery.with {
-            addEntity(Issue)
+
+        final List<Object[]> results = sqlQuery.with {
             setParameterList('issueIds', issueIds)
             list()
         }
-        results
+
+        def rows = results.collect { resultRow ->
+            [
+                id            : resultRow[0],
+                projectKey    : resultRow[1],
+                type          : resultRow[2],
+                status        : resultRow[3],
+                summary       : resultRow[4],
+                reporter      : resultRow[5],
+                reporterName  : resultRow[6],
+                approvalStatus: resultRow[7],
+                updated       : resultRow[8],
+                expirationDate: resultRow[9],
+                name          : resultRow[10],
+                value         : resultRow[11],
+                displayName   : resultRow[12]
+            ]
+        }
+
+        rows.each { 
+            if (it.projectKey == currentProjectKey) {
+                if (it.type != IssueType.CONSENT_GROUP.name) {
+                    issueSearchItemDTO.setExtraProperty(it.name.toString(), it.value.toString())
+                    issueSearchItemDTO.setAccessContacts(it.name.toString(), it.displayName.toString())
+                }
+            } else {
+                if (currentProjectKey != "") {
+                    resultDTO.add(issueSearchItemDTO)
+                }
+                currentProjectKey = it.projectKey
+                issueSearchItemDTO = new IssueSearchItemDTO(it.toSorted())
+
+                if (it.type != IssueType.CONSENT_GROUP.name) {
+                    issueSearchItemDTO.setExtraProperty(it.name.toString(), it.value.toString())
+                    issueSearchItemDTO.setAccessContacts(it.name.toString(), it.displayName.toString())
+                }
+            }
+            resultDTO.add(issueSearchItemDTO)
```

Comment:
I think all of this logic should be in the `IssueSearchItemDTO`. There should be a method that takes the collection of results and returns a`Set<IssueSearchItemDTO>`. By putting all of the constructor logic in the query service, you lose the ability for `IssueSearchItemDTO` to have control over how it is created/populated/etc and therefore is of little value outside of this query service method.